src/containers/location/providers/ReferralAgencyHeaderContainer.js

```diff
@@ -0,0 +1,148 @@
+// @flow
+
+import React from 'react';
+
+import styled from 'styled-components';
+import { faChevronLeft } from '@fortawesome/pro-light-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Colors } from 'lattice-ui-kit';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { selectReferralAgency } from '../LocationsActions';
+import { ContentOuterWrapper, ContentWrapper } from '../../../components/layout';
+import { HEADER_HEIGHT, HEIGHTS } from '../../../core/style/Sizes';
+import { getTextFnFromState } from '../../../utils/AppUtils';
+import { renderFacilityName } from '../../../utils/DataUtils';
+import { LABELS } from '../../../utils/constants/Labels';
+import { PROVIDERS, STATE } from '../../../utils/constants/StateConstants';
+
+const { LOCATIONS } = STATE;
+
+const { NEUTRAL, PURPLE } = Colors;
+
+const StyledContentOuterWrapper = styled(ContentOuterWrapper)`
+  position: fixed;
+  top: ${HEADER_HEIGHT}px;
+  z-index: 1;
+`;
+
+const StyledContentWrapper = styled(ContentWrapper)`
+  background-color: white;
+  position: relative;
+
+  @media only screen and (min-height: ${HEIGHTS[0]}px) {
+    padding: 10px 25px;
+  }
+
+  @media only screen and (min-height: ${HEIGHTS[1]}px) {
+    padding: 25px;
+  }
+`;
+
+const BackButton = styled.div`
+  align-items: center;
+  color: ${PURPLE.P300};
+  display: flex;
+  flex-direction: row;
+  font-size: 14px;
+  font-weight: 600;
+  text-decoration: none;
+
+  :hover {
+    text-decoration: underline;
+  }
+
+  span {
+    margin-left: 15px;
+  }
+
+  &:hover {
+    cursor: pointer;
+  }
+`;
```

Comment:
I feel like I've seen this before. Please move this to `src/components/buttons` and reuse it.